### ISSUE TITLE
Bug: Searching from the Columns View displays the wrong context menu for files

### ISSUE THREAD

**Issue by lrstolk** [files-community/Files#14058]
### Steps to reproduce

1. Switch to the Columns View
2. Start a search
3. Right click file and notice that the context menu is displaying the wrong options

### Requirements

- Fix issue where the context menu is displaying the wrong options

### Files Version

v3.1.6

### Windows Version

10.0.22631.2715

**Comment by Josh65-2201** [comment]
Thanks for the feedback, 
- Files uses the same system as Windows for search can you check they show in Windows File Explorer
- Open File location is already in the context menu on search results
- There is no reason to disable open in new tab as it work for search results too.
-  Can you send a screenshot of the wrong icons in the search bar and what they should be.

**Comment by yaira2** [comment]
> When I press open file in new tab, I get an a logical error. If it gives this error, why even enabling this option in the context menu?

Where do you see this option? It should only be available for folders.

**Comment by lrstolk** [comment]
> Thanks for the feedback,
> 
> * Files uses the same system as Windows for search can you check they show in Windows File Explorer

The results in windows file explorer do show them, so it is something with Files:  
<img width="592" alt="image" src="https://github.com/files-community/Files/assets/35999180/8a278f66-2977-4e04-9971-43290ced12f7"> 


> * Open File location is already in the context menu on search results
> * There is no reason to disable open in new tab as it work for search results too.

> Where do you see this option? It should only be available for folders.

look at my video at [0:28](https://github.com/files-community/Files/assets/35999180/d99f984f-2869-4954-9757-5353be3f4bee). 
<img width="323" alt="image" src="https://github.com/files-community/Files/assets/35999180/6a1c3c3d-6e06-4a1a-9f1e-4d88c93bd592">
There is not.  


> * Can you send a screenshot of the wrong icons in the search bar and what they should be.

What do you mean? It it shown in the video, please check it. 
At [0:06](https://github.com/files-community/Files/assets/35999180/d99f984f-2869-4954-9757-5353be3f4bee)  
<img width="163" alt="image" src="https://github.com/files-community/Files/assets/35999180/5aeb3aa5-5a31-4a4f-92e3-fe7bf390fccb">
I get these results. When I do it again at [0:17](https://github.com/files-community/Files/assets/35999180/d99f984f-2869-4954-9757-5353be3f4bee)  
<img width="177" alt="image" src="https://github.com/files-community/Files/assets/35999180/5f838db1-0d11-441b-8298-c336746467d4"> the results are wrong. 
It is the same search. 
I searched, went to the folder, went back and searched again. The second time it bugged.

**Comment by yaira2** [comment]
Regarding the "open in new tab issue", can you open a bug report? It should be an easy fix but I want to make sure we have a separate issue so we don't lose track.

**Comment by yaira2** [comment]
@lrstolk is the location you're searching indexed by Windows? The default setting in Windows is pretty limited so that would explain why Files isn't showing the expected results. You can solve this by switching to "Enhanced" or by adding the location to the search index.
![image](https://github.com/files-community/Files/assets/39923744/87e9d4a5-019d-46fe-a1d2-deb49e873160)

**Comment by yaira2** [comment]
Is this still an issue in the latest preview version? https://files.community/download/preview

**Comment by lrstolk** [comment]
> @lrstolk is the location you're searching indexed by Windows? The default setting in Windows is pretty limited so that would explain why Files isn't showing the expected results. You can solve this by switching to "Enhanced" or by adding the location to the search index. ![image](https://private-user-images.githubusercontent.com/39923744/294281490-87e9d4a5-019d-46fe-a1d2-deb49e873160.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0OTY3MDEsIm5iZiI6MTcwNTQ5NjQwMSwicGF0aCI6Ii8zOTkyMzc0NC8yOTQyODE0OTAtODdlOWQ0YTUtMDE5ZC00NmZlLWExZDItZGViNDllODczMTYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDEzMDAwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzY2JmMzQxMzM1MmMyMmJhODg4NDYwMDNmMGU2ODIzYmViYWVlNjZkZjAwZWU3MjdjZjlhMDViZWZmNTdkZmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RtbkLJPka-IBNkg9h2pWu6sI0aH1jOUWo0VxYnXjw2A)

Well it shows the right results the first time I search, the first tiem I search everything goed well, but not the second time it shows the wrong results in the preview and things as the context manu start to bug.

**Comment by lrstolk** [comment]
> When you start the second search, are you going back to the original location or are you just changing the text in the search box and pressing enter?

I go back, so I search by typing "238", press enter, go back, typing"238", press enter. 
Also the main problem is not solved, I got the latest version from 14 januari 2024. 

Also check this videos to see my issues: 
https://github.com/files-community/Files/assets/35999180/e8cc8dbb-d3c2-4741-92e8-5fa01faea1d4

**Comment by yaira2** [comment]
It's hard to follow the video but if you can summarize steps to reproduce the issue I'll see if I can repro it on my end. It might also be worth checking if you can repro this issue on a fresh install of Files without making any changes to the settings or layout modes.

**Comment by yaira2** [comment]
@lrstolk are you by any chance starting the search while using the Columns Layout? If so, it's probably the same issue as #13311 and I'm able to reproduce the issue as well.

**Comment by lrstolk** [comment]
> It's hard to follow the video but if you can summarize steps to reproduce the issue I'll see if I can repro it on my end. It might also be worth checking if you can repro this issue on a fresh install of Files without making any changes to the settings or layout modes.

What can I make more clear for you? Cause for me it is straight forward.

**Comment by lrstolk** [comment]
> @lrstolk are you by any chance starting the search while using the Columns View? If so, it's probably the same issue as #13311 and I'm able to reproduce the issue as well.

I am, but as I've read that issue #13311, it is not the same.

**Comment by yaira2** [comment]
Now that I'm able to reproduce the issue, I'll edit the description for future reference.

**Comment by yair100** [comment]
A fix will be included in the next update.
